src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -98,19 +98,18 @@ class BoringUtilsException(message: String) extends Exception(message)
   */
 object BoringUtils {
   /* A global namespace for boring ids */
-  private val namespace: SyncVar[Namespace] = new SyncVar
-  namespace.put(Namespace.empty)
+  private[chisel3] case object CacheKey extends BuilderContextCache.Key[Namespace]
+  private val boringNamespace = Builder.contextCache.getOrElseUpdate(CacheKey, Namespace.empty)
```

Comment:
```suggestion
  private def boringNamespace = Builder.contextCache.getOrElseUpdate(CacheKey, Namespace.empty)
```
This cannot be a `val`, the point is that multiple Chisel elaborations need to be able to run in the same JVM at the same time.